otx/algorithms/classification/adapters/mmcls/configurer.py

```diff
@@ -0,0 +1,701 @@
+"""Base configurer for mmdet config."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import importlib
+import json
+import os
+from typing import Any, Dict
+
+import numpy as np
+import torch
+from mmcv import build_from_cfg
+from mmcv.runner import CheckpointLoader
+from mmcv.utils import Config, ConfigDict
+from torch import distributed as dist
+
+from otx.algorithms import TRANSFORMER_BACKBONES
+from otx.algorithms.classification.adapters.mmcls.utils import (
+    patch_datasets,
+    patch_evaluation,
+)
+from otx.algorithms.common.adapters.mmcv.utils import (
+    align_data_config_with_recipe,
+    build_dataloader,
+    build_dataset,
+    patch_adaptive_interval_training,
+    patch_default_config,
+    patch_early_stopping,
+    patch_fp16,
+    patch_persistent_workers,
+    patch_runner,
+)
+from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
+    recursively_update_cfg,
+    update_or_add_custom_hook,
+)
+from otx.algorithms.common.utils.logger import get_logger
+
+logger = get_logger()
+
+
+# pylint: disable=too-many-public-methods
+class ClassificationConfigurer:
+    """Patch config to support otx train."""
+
+    def __init__(self):
+        self.task_adapt_type = None
+        self.task_adapt_op = "REPLACE"
+        self.org_model_classes = []
+        self.model_classes = []
+        self.data_classes = []
+
+    # pylint: disable=too-many-arguments
+    def configure(
+        self,
+        cfg,
+        model_ckpt,
+        data_cfg,
+        training=True,
+        subset="train",
+        ir_options=None,
+        data_classes=None,
+        model_classes=None,
+        **kwargs,
+    ):
+        """Create MMCV-consumable config from given inputs."""
+        logger.info(f"configure!: training={training}")
+
+        self.configure_base(cfg, data_cfg, data_classes, model_classes, **kwargs)
+        self.configure_device(cfg, training)
+        self.configure_ckpt(cfg, model_ckpt)
+        self.configure_model(cfg, ir_options)
+        self.configure_data(cfg, training, data_cfg)
+        self.configure_task(cfg, training)
+        self.configure_hook(cfg)
+        self.configure_samples_per_gpu(cfg, subset)
+        self.configure_fp16_optimizer(cfg)
+        self.configure_compat_cfg(cfg)
+        return cfg
+
+    def configure_base(self, cfg, data_cfg, data_classes, model_classes, **kwargs):
+        """Basic configuration work for recipe.
+
+        Patchings in this function are handled task level previously
+        This function might need to be re-orgianized
+        """
+
+        options_for_patch_datasets = kwargs.get("options_for_patch_datasets", {"type": "OTXClsDataset"})
+        options_for_patch_evaluation = kwargs.get("options_for_patch_evaluation", {"task": "normal"})
```

Comment:
Is these options changed by type of classification?

Reply:
Yes, I've kept it as is for now (Stage..?). If I get a chance, I'd like to fix the multi-class, multi-label, and h-label patching methods.